promql/functions.go

```diff
@@ -814,11 +863,11 @@ var functions = map[string]*Function{
 		Call:       funcResets,
 	},
 	"round": {
-		Name:         "round",
-		ArgTypes:     []model.ValueType{model.ValVector, model.ValScalar},
-		OptionalArgs: 1,
-		ReturnType:   model.ValVector,
-		Call:         funcRound,
+		Name:       "round",
+		ArgTypes:   []model.ValueType{model.ValVector, model.ValScalar},
+		Variadic:   true,
```

Comment:
Well yes, this being a static error is what I want, no matter the comparison to Python or not :) So that sounds like you share my opinion of adding an optional limit for variadic args at least later on to (re)introduce this kind of checking.